### PR TITLE
add URLPath in the connection options

### PR DESCRIPTION
**Pull request by gufranmirza**

### CODE REVIEW

File: fluent/client/client.go [hunk 1]
```diff
@@ -43,12 +43,18 @@ type Client struct {
...
-	Hostname string
-	Port     int
+	Hostname string // Hostname of the server to connect
+	Port     int    // Port of the server
+	URLPath  string // Optional Path to connect on server, it should start with trailing slash e.g. /events/ingest
 }
 
-func (sa ServerAddress) String() string {
+func (sa *ServerAddress) String() string {
+	if len(sa.URLPath) != 0 {
```
Comment (ericmillin) [inline fluent/client/client.go:54]: Port isn't required for websockets. Update comment to clarify. Then do something like:

```go
func (sa ServerAddress) String() string {
        hn :=  sa.Hostname

	if sa.Port != 0 {
	      hn = fmt.Sprintf("%s:%d", hn, sa.Port)
	}

	return path.Join(hn, sa.Path)
}
```

 Note that `String()` doesn't need a pointer receiver

File: cmd/ws/main.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ func main() {
 		ServerAddress: client.ServerAddress{
 			Hostname: "ws://127.0.0.1",
 			Port:     8083,
+			URLPath:  "/events/ingest", // This path is optional
```
[the remark this answers is not in the document]
Comment (chiragkyal) [reply cmd/ws/main.go]: Done

File: fluent/client/client.go [hunk 3]
```diff
@@ -43,12 +43,18 @@ type Client struct {
 	Hostname string
 }
 
+// ServerAddress holds details about URLs of the server to connect
 type ServerAddress struct {
-	Hostname string
-	Port     int
+	Hostname string // Hostname of the server to connect
+	Port     int    // Port of the server
+	URLPath  string // Optional Path to connect on server, it should start with trailing slash e.g. /events/ingest
```
[the remark this answers is not in the document]
Comment (chiragkyal) [reply fluent/client/client.go]: Done

File: fluent/client/client.go (hunk 1 above)
Comment (chiragkyal) [reply fluent/client/client.go]: Updated

**Review by ericmillin — APPROVED** [review]
lgtm

**Pull request merged**